### PR TITLE
Distributed Transaction Resolver

### PR DESCRIPTION
**Pull request by harshit-gangal**
## Description

This PR changes VTGate to watch on transaction unresolved signal from the health stream. When received it invokes transaction resolution for that `keyspace:shard`. The transaction resolver retrieves the unresolved transaction list and resolves those transactions based on the current state i.e. commit/rollback/prepare.

## Related Issue(s)

- #16245 

## Checklist

-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by GuptaManan100 — APPROVED** [review]
LGTM!

File: go/vt/vtgate/tx_conn.go [hunk 1]
```diff
@@ -201,25 +201,51 @@ func (txc *TxConn) commit2PC(ctx context.Context, session *SafeSession) error {
 		return err
 	}
 
+	// retrieve the caller ID to execute the testing flow
+	callerID := callerid.EffectiveCallerIDFromContext(ctx)
+
+	// Test code to simulate a failure after RM prepare
+	if failNow, err := checkTestFailure(callerID, "TRCreated_FailNow", nil); failNow {
```
Comment (systay) [inline go/vt/vtgate/tx_conn.go:208]: I'm confused. Isn't this test code? What is it doing here?

Comment (harshit-gangal) [reply go/vt/vtgate/tx_conn.go:208]: We will add the `go build` logic to include test and non-test functions. There is already a task to make the changes on the commit flow. That will be added as part of the PR.

Comment (GuptaManan100) [reply go/vt/vtgate/tx_conn.go]: Done! In the latest commits! 🚀

**Review by systay — APPROVED** [review]

**Pull request merged**